lib/semantics/check-omp-structure.cc

```diff
@@ -881,8 +909,41 @@ void OmpStructureChecker::Enter(const parser::OmpLinearClause &x) {
     }
   }
 }
-void OmpStructureChecker::Enter(const parser::OmpMapClause &) {
+void OmpStructureChecker::Enter(const parser::OmpMapClause &x) {
   CheckAllowed(OmpClause::MAP);
+  if (const auto &maptype{std::get<std::optional<parser::OmpMapType>>(x.t)}) {
+    using Type = parser::OmpMapType::Type;
+    const Type &type = std::get<Type>(maptype->t);
+    switch (GetContext().directive) {
+    case OmpDirective::TARGET:
+    case OmpDirective::TARGET_DATA: {
+      if (type != Type::To && type != Type::From && type != Type::Tofrom &&
+          type != Type::Alloc) {
+        context_.Say(GetContext().clauseSource,
+            "Only the to, from, tofrom or alloc map types are permitted "
+            "for MAP clauses on the %s directive"_err_en_US,
```

Comment:
I think it would be better to capitalize "to", "from", etc. in these messages because they are keywords, like MAP.